src/web/services/student.service.spec.ts

```diff
@@ -82,18 +84,59 @@ describe('StudentService', () => {
         .toHaveBeenCalledWith(ResourceEndpoints.STUDENT_COURSE_LINKS_REGENERATION, paramMap);
   });
 
-  it('should execute GET when loading students in a course as CSV', () => {
-    const paramMap: Record<string, string> = {
-      courseid: 'CS3281',
-    };
+  it('should show course student list with section', () => {
+    const course: string = 'typicalWithSection';
+    spyOn(spyHttpRequestService, 'get').and.callFake((endpoint: ResourceEndpoints, params: Record<string, string>) => {
+      if (endpoint === ResourceEndpoints.COURSE) {
+        expect(params).toEqual({
+          courseid: course,
+          entitytype: 'instructor',
+        });
+        return of({
+          courseId: course,
+          courseName: course,
+          timeZone: 'UTC',
+          creationTimestamp: 123,
+          deletionTimestamp: 123,
+        });
+      }  if (endpoint === ResourceEndpoints.STUDENTS) {
+        return of({
+          students: [{
+            email: 'student1OfTypicalCourse@typicalCourse.com',
+            courseId: course,
+            name: 'student1OfTypicalCourse',
+            joinState: JoinState.JOINED,
+            teamName: 'Team 1',
+            sectionName: 'Section A',
+          },
+          {
+            email: 'student2OfTypicalCourse@typicalCourse.com',
+            courseId: course,
+            name: 'student2OfTypicalCourse',
+            joinState: JoinState.JOINED,
+            teamName: 'Team 2',
+            sectionName: 'Section A',
+          }],
+        });
+      }
+      return null;
+    });
+    expect(service.loadStudentListAsCsv({ courseId: course })).toMatchSnapshot();
```

Comment:
Mistakes:
- You're setting the spy here on the test method, but the service to be tested has been injected with the default `spyHttpRequestService` (with none of the above having any effect) way up in the `beforeEach` block.
- Here `toMatchSnapshot` is called on the `Observable`, not the CSV string itself. It won't produce the testing effect we want to achieve.

Suggestion:

While setting a spy to the `HttpRequestService` works, it will be troublesome to set up multiple different spies for different test cases. So why not:
- In `student.service.ts`, create a method that takes in `Students` and outputs CSV string.
- Test that method extensively. You don't need any spy to do that, and you also don't need to worry about `Observable`s.